agent.go

```diff
@@ -28,7 +29,11 @@ var icp0, _ = url.Parse("https://icp0.io/")
 
 func effectiveCanisterID(canisterId principal.Principal, args []any) principal.Principal {
 	// If the canisterId is not aaaaa-aa, return it.
-	if len(canisterId.Raw) > 0 || len(args) == 0 {
+	if canisterId.Encode() != ic.MANAGEMENT_CANISTER_PRINCIPAL.Encode() {
+		return canisterId
+	}
```

Comment:
We can just check whether the `canisterId` is an empty slice, instead of converting it to a string.

Reply:
Right, I thought this was clearer but I can just check for the empty slice and leave a note (there is already a note but it didn't make sense to me until @frederikrothenberger mentioned aaaaa-aa was 0 bytes)